templates/watcher/config/00-default.conf

```diff
@@ -0,0 +1,48 @@
+[DEFAULT]
+state_path = /var/lib/watcher
+transport_url = {{ .TransportURL }}
+control_exchange = watcher
+debug = True
+
+[database]
+connection = {{ .DatabaseConnection }}
+
+[oslo_policy]
+policy_file = /etc/watcher/policy.yaml.sample
+
+[oslo_messaging_notifications]
+driver = messagingv2
+
+[keystone_authtoken]
+memcached_servers = {{ .MemcachedServers }}
+# TODO jgilaber implement handling this option when we add tls support
+# cafile = /var/lib/ca-bundle.pem
+project_domain_name = Default
+project_name = service
+user_domain_name = Default
+password = {{ .ServicePassword }}
+username = {{ .ServiceUser }}
+auth_url = {{ .KeystoneAuthURL }}
+interface = internal
+auth_type = password
+
+[watcher_clients_auth]
+# TODO jgilaber implement handling this option when we add tls support
+# cafile = /var/lib/ca-bundle.pem
+project_domain_name = Default
+project_name = service
+user_domain_name = Default
+password = {{ .ServicePassword }}
+username = {{ .ServiceUser }}
+auth_url = {{ .KeystoneAuthURL }}
+interface = internal
+auth_type = password
+
+[oslo_concurrency]
+lock_path = /var/lib/watcher/tmp
+
+[watcher_datasources]
+datasources = ceilometer
```

Comment:
actually we should not merge with this as Takashi is actually removing with https://review.opendev.org/c/openstack/watcher/+/937625

really we want the default to be prometheus? it is up to the deployer what the default is here i think.

Reply:
best thing is to leave empty, so strategies will check for all available datasources.

Reply:
+1 to go with ceilometer now.

Reply:
WRT:

> really we want the default to be prometheus? it is up to the deployer what the default is here i think.

I think we should leave a meaningful default value. In this case, the deployer using openstack-k8s-operators have no option to deploy monasca, gnocchi and will be limited to prometheus, so I think we should default to prometheus.

> best thing is to leave empty, so strategies will check for all available datasources.

What I've seen in logs is that, when all the datasources are tried, it tries multiple times (10 by default) with a certain timeout on each one. I think this is a waste of resources for a case where we know (or we will know) that prometheus is really the only available data source (as deployed by operators). So, I'd be prescriptive in the default configuration.

In any case, we are still not running any service and we will have follow-up PRs where we will do the actual deployment.  We can retake this discussion by then.